Shape_regularization/include/CGAL/Shape_regularization/internal/Contour_base_2.h

```diff
@@ -752,7 +752,7 @@ namespace internal {
       const Line_2& line_2,
       Point_2& in_point) const {
 
-      typename std::result_of<Intersect_2(Line_2, Line_2)>::type result
+      typename CGAL::cpp11::result_of<Intersect_2(Line_2, Line_2)>::type result
```

Comment:
Is there any advantage over `auto`?
```suggestion
      auto result
```